server/methods.js

```diff
@@ -215,38 +215,69 @@ Meteor.methods({
 
   */
   flickrImageSearchList: function(query, option, page) {
-    var items, nextPage;
+    var items, nextPage, keywordSearch, path, requestParams;
     check(query, String);
+
+    if ((query.indexOf('flickr.com') !==-1) && (query.indexOf('/photos/') !==-1)){
+      keywordSearch = false;
+      //search photo: flickr.com/photos/{user-id}/{photo-id}/in/photolist-{search-info}
+      //individual photo:  flickr.com/photos/{user-id}/{photo-id}
+      var split = _.chain(query.split('/')).compact().value();
+      var offset = split.indexOf('photos');
+      if (split[offset+2]) var photo_id = (split[offset+2]).match(/[\d]*/)[0];
+    } else if ((query.indexOf('flic.kr') !==-1) && (query.indexOf('/p/') !==-1)){
+      keywordSearch = false;
+      //short url: https://flic.kr/p/{base58-photo-id}
+      var photo_id = _.chain(query.split('/')).compact().last().value().match(/[\d\w]*/)[0];
+    } else {
+      keywordSearch = true;
+    }
+
     page = page || 1;  // flickr starts from 1
     this.unblock();
 
-    var url = "https://api.flickr.com/services/rest/?&method=flickr.photos.search";
-
-    var requestParams = {
-      tags: query.replace(' ', ','),
-      text: query,
-      api_key: FLICKR_API_KEY,
-      format: 'json',
-      privacy_filter: 1,
-      media: 'photos',
-      nojsoncallback: 1,
-      sort: 'relevance',
-      license: '1,2,3,4,5,6,7,8',
-      per_page: 200,
-      extras: ['owner_name', 'date_upload'],
-      page: page
-    };
+    if (keywordSearch) {
+      path = 'flickr.photos.search';
+      requestParams = {
+        tags: query.replace(' ', ','),
+        text: query,
+        api_key: FLICKR_API_KEY,
+        format: 'json',
+        privacy_filter: 1,
+        media: 'photos',
+        nojsoncallback: 1,
+        sort: 'relevance',
+        license: '1,2,3,4,5,6,7,8',
+        per_page: 200,
+        extras: ['owner_name', 'date_upload'],
+        page: page
+      };
+    } else if (photo_id) {
+      path = 'flickr.photos.getInfo';
+      requestParams = {
+        photo_id: photo_id,
+        api_key: FLICKR_API_KEY,
+        format: 'json',
+        nojsoncallback: 1,
+      };
+    } 
+
+    var url = "https://api.flickr.com/services/rest/?&method=" + path;
 
     var res = HTTP.get(url, {
       params: requestParams
     });
 
-    if (res.content){
-      items = JSON.parse(res.content).photos.photo;
+    if (res.data) var results = res.data; 
```

Comment:
otherwise the variable `results` is never declared, so will hit an error below. i think using braces and a separate line for the `if` and the code path makes this clearer.